pkg/note/note.go

```diff
@@ -150,3 +168,43 @@ func NewNoteSigner(ctx context.Context, origin string, signer signature.Signer)
 		sign: sign,
 	}, nil
 }
+
+func NewNoteVerifier(ctx context.Context, origin string, verifier signature.Verifier) (note.Verifier, error) {
+	if !isValidName(origin) {
+		return nil, fmt.Errorf("invalid name %s", origin)
+	}
+
+	pubKey, err := verifier.PublicKey()
+	if err != nil {
+		return nil, fmt.Errorf("getting public key: %w", err)
+	}
+
+	var keyID uint32
+	switch pk := pubKey.(type) {
+	case *ecdsa.PublicKey:
+		keyID, err = ecdsaKeyHash(pk)
+		if err != nil {
+			return nil, fmt.Errorf("getting ECDSA key hash: %w", err)
+		}
+	case ed25519.PublicKey:
+		keyID = ed25519KeyHash(origin, pk)
```

Comment:
how does this error out?

Reply:
it does not -- it never tries to parse the key, just hashes the bytes